src/entities/EditorDocument/index.ts

```diff
@@ -27,4 +27,57 @@ export class EditorDocument {
     this.#children = children;
     this.#properties = properties;
   }
+
+  /**
+   * Adds a BlockNode to the EditorDocument at the specified index.
+   * If no index is provided, the BlockNode will be added to the end of the array.
+   * Throws an error if the index is out of bounds.
+   *
+   * @param blockNode - The BlockNode to add to the EditorDocument
+   * @param index - The index at which to add the BlockNode
+   */
+  public addBlock(blockNode: BlockNode, index?: number): void | never {
+    if (index === undefined) {
+      this.#children.push(blockNode);
+
+      return;
+    }
+
+    /**
+     * Throws error if index is out of bounds
+     */
+    if (index < 0 || index > this.#children.length) {
+      throw new Error('Index out of bounds');
+    }
+
+    this.#children.splice(index, 0, blockNode);
+  }
+
+  /**
+   * Removes a BlockNode from the EditorDocument at the specified index.
+   * Throws an error if the index is out of bounds.
+   *
+   * @param index - The index of the BlockNode to remove
+   */
+  public removeBlock(index: number): void | never {
+    if (index < 0 || index >= this.#children.length) {
```

Comment:
You can extract that to a separate method to use in both addBlock and removeBlock